pkg/controller/servicebinding/reconciler_test.go

```diff
@@ -391,10 +407,18 @@ func TestReconcilerUpdateCredentials(t *testing.T) {
 	r := &reconciler{dynClient: fakeDynClient, restMapper: mapper, scheme: f.S}
 	r.resourceWatcher = newFakeResourceWatcher(mapper)
 
-	u, err := fakeDynClient.Resource(secretsGVR).Get("db-credentials", metav1.GetOptions{})
+	res, err := r.Reconcile(reconcileRequest())
+	require.NoError(t, err)
+	require.False(t, res.Requeue)
+
+	namespacedName := types.NamespacedName{Namespace: reconcilerNs, Name: reconcilerName}
+	sbrOutput1, err := r.getServiceBinding(namespacedName)
+	require.NoError(t, err)
+
+	bindingSecret1, err := fakeDynClient.Resource(secretsGVR).Get(sbrOutput1.Status.Secret, metav1.GetOptions{})
```

Comment:
The original version the secret subject to change was the one associated with the service, not the Service Binding one. This change removed the coverage for situations a secret related to the service has been changed, so I would urge to reconsider and create a new test for the functionality you're trying to test here.